CMakeLists.txt

```diff
@@ -330,6 +334,7 @@ if (NOT WIN32)
 endif ()
 
 add_library(freeorioncommon
+    $<$<CXX_COMPILER_ID:msvc>:STATIC>
```

Comment:
> The repo project files were based on (possibly older) CMake-generated project files.

That's not true.  The repo history shows that.  Also the generated projects are unfit for source tracking.

@o01eg Don't waste your time.  Making the libaries shared isn't as easy on Windows as it is on the other two platforms as there are circular symbol dependencies that need to be sorted out before whereas the other platforms have a linker flag to let the runtime linker resolve those.